File: proposed-projects/round-trip-experiment/reverse/20260909T185402Z_14873f98/20260910T102153Z_a4bbfe64/ported_implementation/src/utils/errors/getInputAsString.ts
import type { ValidInput } from '../../grammarGraph/types';

export const getInputAsString = (src: ValidInput): string => {
  if (typeof src === 'string') {
    return src;
  }
  const codePoints = Array.isArray(src) ? src : [src];
  return codePoints.map((cp) => String.fromCodePoint(cp)).join('');
};

File: packages/gbnf-experiment/data/20260909T133702Z_8305aa82/ported_implementation/gbnf/utils/errors/get_input_as_string.py
from ...grammar_graph.types import ValidInput


def get_input_as_string(src: ValidInput) -> str:
    if isinstance(src, str):
        return src
    code_points = src if isinstance(src, (list, tuple)) else [src]
    return "".join(chr(cp) for cp in code_points)
